### PR TITLE
Migrate Savestates Home Menu

### PR DESCRIPTION
**Pull request by Darkhost1999**
In preparation for Savestate Slots move all the current code to a subpage of Savestates

Screenshot 2024-11-22 164446
Screenshot 2024-11-22 164459

Functionality remains intact, Just simply moving the current code to a dedicated file and creating a new page then applying the layout inside of the new page.
Also a misc change with building.md to update the information to msvc2022 instead of 2019 post QT 6.8.0 merge.
Another misc change I was saving for a bigger PR like this to revert a bad decision that didn't work.

The thought about future functionality to add coming soon is Savestate Slot 1, Savestate Slot 2, and Savestate Slot 3. Then Reload Savestate Slot 1, Reload Savestate Slot 2, and Reload  Savestate Slot 3.
Why 3 slots?
All the games I have observed have up to 3 games. Kingdom Hearts launchers have 2 games + 1 cinema, Sly Cooper trilogy has 3 games, Devil May Cry HD Collection has 3 games. If anybody has any feedback about a game that has 4 or more games in its launcher please let me know. Journey Collector's Edition doesn't count because the games are installed individually into RPCS3 and can be launched separately without issue.

### CODE REVIEW

File: rpcs3/Emu/RSX/Overlays/HomeMenu/overlay_home_menu_main_menu.h [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 #include "overlay_home_menu_page.h"
 #include "overlay_home_menu_settings.h"
+#include "overlay_home_menu_savestate.h"
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/Overlays/HomeMenu/overlay_home_menu_main_menu.h:5]: Why is this included here?

File: rpcs3/Emu/RSX/Overlays/HomeMenu/overlay_home_menu_main_menu.cpp [hunk 2]
```diff
@@ -8,8 +8,6 @@
 
 extern atomic_t<bool> g_user_asked_for_recording;
 extern atomic_t<bool> g_user_asked_for_screenshot;
-extern bool boot_last_savestate(bool testing);
-
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/Overlays/HomeMenu/overlay_home_menu_main_menu.cpp:12]: Missing newline

File: rpcs3/Emu/RSX/Overlays/HomeMenu/overlay_home_menu_main_menu.h (hunk 1 above)
Comment (Darkhost1999) [reply rpcs3/Emu/RSX/Overlays/HomeMenu/overlay_home_menu_main_menu.h:5]: I saw settings.h and thought that would be the best place for savestate.h for the build to compile. else I get
```
3>C:\Users\bryso\Documents\GitHub\rpcs3\rpcs3\Emu\RSX\Overlays\HomeMenu\overlay_home_menu_main_menu.cpp(76,30): error C2065: 'home_menu_savestate': undeclared identifier
3>(compiling source file '/Emu/RSX/Overlays/HomeMenu/overlay_home_menu_main_menu.cpp')
3>C:\Users\bryso\Documents\GitHub\rpcs3\rpcs3\Emu\RSX\Overlays\HomeMenu\overlay_home_menu_main_menu.cpp(76,18): error C2672: 'std::make_shared': no matching overloaded function found
3>(compiling source file '/Emu/RSX/Overlays/HomeMenu/overlay_home_menu_main_menu.cpp')
3>    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34604\include\memory(2959,44):
3>    could be 'std::shared_ptr<_Ty> std::make_shared(const remove_extent<_Ty>::type &)'
3>        C:\Users\bryso\Documents\GitHub\rpcs3\rpcs3\Emu\RSX\Overlays\HomeMenu\overlay_home_menu_main_menu.cpp(76,30):
3>        'std::make_shared': invalid template argument for '_Ty', type expected
3>    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34604\include\memory(2950,44):
3>    or       'std::shared_ptr<_Ty> std::make_shared(void)'
3>        C:\Users\bryso\Documents\GitHub\rpcs3\rpcs3\Emu\RSX\Overlays\HomeMenu\overlay_home_menu_main_menu.cpp(76,30):
3>        'std::make_shared': invalid template argument for '_Ty', type expected
3>    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34604\include\memory(2945,44):
3>    or       'std::shared_ptr<_Ty> std::make_shared(const size_t,const remove_extent<_Ty>::type &)'
3>        C:\Users\bryso\Documents\GitHub\rpcs3\rpcs3\Emu\RSX\Overlays\HomeMenu\overlay_home_menu_main_menu.cpp(76,30):
3>        'std::make_shared': invalid template argument for '_Ty', type expected
3>    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34604\include\memory(2940,44):
3>    or       'std::shared_ptr<_Ty> std::make_shared(const size_t)'
3>        C:\Users\bryso\Documents\GitHub\rpcs3\rpcs3\Emu\RSX\Overlays\HomeMenu\overlay_home_menu_main_menu.cpp(76,30):
3>        'std::make_shared': invalid template argument for '_Ty', type expected
3>    C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34604\include\memory(2903,44):
3>    or       'std::shared_ptr<_Ty> std::make_shared(_Types ...)'
3>        C:\Users\bryso\Documents\GitHub\rpcs3\rpcs3\Emu\RSX\Overlays\HomeMenu\overlay_home_menu_main_menu.cpp(76,30):
3>        'std::make_shared': invalid template argument for '_Ty', type expected
3>Done building project "emucore.vcxproj" -- FAILED.
```

Comment (Megamouse) [reply rpcs3/Emu/RSX/Overlays/HomeMenu/overlay_home_menu_main_menu.h:5]: Obviously you have to include it in the cpp, not the h.
You can probably also move the settings.h

Comment (Darkhost1999) [reply rpcs3/Emu/RSX/Overlays/HomeMenu/overlay_home_menu_main_menu.h]: Done moved both the settings.h and savestate.h to the .cpp

**Comment by Jack54guythecoder** [comment]
> If anybody has any feedback about a game that has 4 or more games in its launcher please let me know. 


The Orange Box has 5 games, but ps3 ver is not so good.
